src/k8s_resource_handler/kubernetes/_kubernetes_resource_handler.py

```diff
@@ -70,70 +90,115 @@ def compute_unit_status(
-
         Returns: A charm unit status (one of ActiveStatus, WaitingStatus, or BlockedStatus)
         """
         self.log.info("Computing a suggested unit status describing these Kubernetes resources")
 
-        if resources is None:
-            resources = self.render_manifests()
-
+        resources = self.render_manifests()
         resources_ok, errors = check_resources(self.lightkube_client, resources)
-
         suggested_unit_status = self._charm_status_given_resource_status(resources_ok, errors)
 
-        self.log.info(
+        self.log.debug(
             "Returning status describing Kubernetes resources state (note: this status "
             f"is not applied - that is the responsibility of the charm): {suggested_unit_status}"
         )
         return suggested_unit_status
 
-    def reconcile(self, resources: Optional[LightkubeResourcesList] = None):
+    def reconcile(self):
         """To be implemented. This will reconcile resources, including deleting them."""
         raise NotImplementedError()
 
-    def render_manifests(self) -> LightkubeResourcesList:
-        """Renders this charm's manifests, returning them as a list of Lightkube Resources."""
+    def render_manifests(
+        self,
+        template_files: Optional[Iterable[str]] = None,
+        context: Optional[dict] = None,
+        force_recompute: bool = False,
+    ) -> LightkubeResourcesList:
+        """Renders this charm's manifests, returning them as a list of Lightkube Resources.
+
+        This method requires that template_files and context are both set.
+
+        Args:
+            template_files (iterable): (Optional) If provided, will replace existing value stored
+                                       in self.template_files.
+                                       This is a convenience provided to make the commonly used
+                                       `self.context = context; self.render_manifests()` more
+                                        convenient.
+            context (dict): (Optional) If provided, will replace existing value stored in
+                            self.context.  This is a convenience provided to make the commonly
+                            used `self.context = context; self.render_manifests()` more convenient.
+            force_recompute (bool): If true, will always recompute manifests even if cached
+                                    manifests are available
+        """
         self.log.info("Rendering manifests")
-        context = self._context_factory()
-        template_files = self._template_files_factory()
 
-        self.log.debug(f"Rendering with context: {context}")
+        # Update inputs
+        if template_files is not None:
+            self.template_files = template_files
+        if context is not None:
+            self.context = context
+
+        # Return from cache if available
+        if self._manifests is not None and force_recompute is False:
```

Comment:
To clarify, I think you're asking if it makes sense to cache the `_manifests`, not whether it makes sense to have a `force_recompute` option.  But if I'm mistaken, please correct me :) 

Yeah, I *think* it is feasible and important.  The case I'm thinking of is:
```python
def reconcile():
  self.krh.render_manifests()  # (could leave this out since .apply will do this implicitly, but just to make it clear...
  self.krh.apply()
  self.krh.delete_unnecessary_resources()
```

Without the cache (or passing resources as args), the three commands could work on different manifests, but with the cache this works as someone would expect (compute on command one, use those for commands 2 and 3).

What do you think?

Reply:
Yes, I'm talking about caching `_manifests`. I didn't think this would be used that way, seems a bit weird to have something that will be used by a method that's not implemented yet. I was thinking if it'd work given the charm code will always have `_manifests` as None on every event that's triggered and handled by the charm code.